### PR TITLE
[#171821415] Fix QRCode minlength

### PR DESCRIPTION
**Pull request by BurnedMarshal**

### CODE REVIEW

File: src/__tests__/pagopa.test.ts [hunk 1]
```diff
@@ -133,25 +133,92 @@ describe("PaymentNoticeNumberFromString", () => {
...
+    async (
+      qrCodeSrt,
+      paymentNoticeNumber,
+      expectedAmountLegth,
+      expectedAuxDigit
+    ) => {
+      const validation = PaymentNoticeQrCodeFromString.decode(qrCodeSrt);
+      expect(isRight(validation)).toBeTruthy();
+      if (isRight(validation)) {
+        expect(validation.value.amount).toHaveLength(expectedAmountLegth);
```
Comment (Undermaken) [inline src/__tests__/pagopa.test.ts]:
```suggestion
        expect(validation.value.amount).toHaveLength(expectedAmountLength);
```

File: src/__tests__/pagopa.test.ts [hunk 2]
```diff
@@ -133,25 +133,92 @@ describe("PaymentNoticeNumberFromString", () => {
...
+      4,
+      "3",
+      "should succeed with valid QrCode of four digits amount"
+    ]
+  ])(
+    "%s, %s, %s, %s",
+    async (
+      qrCodeSrt,
+      paymentNoticeNumber,
+      expectedAmountLegth,
```
Comment (Undermaken) [inline src/__tests__/pagopa.test.ts]:
```suggestion
      expectedAmountLength,
```

File: src/__tests__/pagopa.test.ts [hunk 3]
```diff
@@ -133,25 +133,92 @@ describe("PaymentNoticeNumberFromString", () => {
...
-      expect(validation.value.paymentNoticeNumber.auxDigit).toEqual("1");
-      // tslint:disable-next-line:no-any
-      expect((validation.value.paymentNoticeNumber as any).iuv17).toEqual(
-        "23456789012345678"
-      );
-      expect(PaymentNoticeQrCodeFromString.encode(validation.value)).toEqual(
-        qrCodeSrt
-      );
+  it.each([
+    [
```
Comment (Undermaken) [inline src/__tests__/pagopa.test.ts]: The tests fail by adding this test

```
[
      "PAGOPA|002|223456789012345678|12345678901|1234567801",
      "234567890123456",
      10,
      "2",
      "should succeed with valid QrCode"
    ]
```
Could we consider to test also the `aux 2` case?

```typescript
else if (expectedAuxDigit === "2") {
          expect(
            (validation.value.paymentNoticeNumber as PaymentNoticeNumber2).iuv15
          ).toEqual(paymentNoticeNumber);
        }
```

Comment (BurnedMarshal) [reply src/__tests__/pagopa.test.ts]: Fixed in 3273538

File: src/__tests__/pagopa.test.ts (hunk 2 above)
Comment (BurnedMarshal) [reply src/__tests__/pagopa.test.ts]: Fixed in 3273538

File: src/__tests__/pagopa.test.ts (hunk 1 above)
Comment (BurnedMarshal) [reply src/__tests__/pagopa.test.ts]: Fixed in 3273538

File: src/__tests__/pagopa.test.ts [hunk 4]
```diff
@@ -133,25 +133,92 @@ describe("PaymentNoticeNumberFromString", () => {
...
+        );
+
+        if (expectedAuxDigit === "0" || expectedAuxDigit === "3") {
+          // tslint:disable-next-line:no-any
+          expect((validation.value.paymentNoticeNumber as any).iuv13).toEqual(
+            paymentNoticeNumber
+          );
+        } else {
+          // tslint:disable-next-line:no-any
+          expect((validation.value.paymentNoticeNumber as any).iuv17).toEqual(
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply src/__tests__/pagopa.test.ts]: Fixed in 3273538

File: src/__tests__/pagopa.test.ts [hunk 5]
```diff
@@ -133,25 +133,92 @@ describe("PaymentNoticeNumberFromString", () => {
...
+        expect(validation.value.identifier).toHaveLength(6);
+        expect(validation.value.version).toHaveLength(3);
+        expect(validation.value.organizationFiscalCode).toHaveLength(11);
+        expect(validation.value.paymentNoticeNumber.auxDigit).toEqual(
+          expectedAuxDigit
+        );
+
+        if (expectedAuxDigit === "0" || expectedAuxDigit === "3") {
+          // tslint:disable-next-line:no-any
+          expect((validation.value.paymentNoticeNumber as any).iuv13).toEqual(
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply src/__tests__/pagopa.test.ts]: Fixed in 3273538

**Pull request merged**